File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let prevslow= null;
    let slow = head;
    let fast = head;
    for(let i=0;i<n;i++){
        fast = fast.next
    }
    if (fast==null){
        // means pahle node hi remove ho gaya hai
        return head.next
    }
    while(fast!=null){
        fast = fast.next;
        prevslow= slow;
        slow = slow.next;
    }
    prevslow.next = slow.next;
    return head
    
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if (head == NULL || head->next == NULL)
            return NULL;
        ListNode* slow = head;
        ListNode* fast = head;
        ListNode* prevslow = NULL;
        for(int i=1;i<=n;i++){
            fast = fast->next;
        }
        if(fast==NULL) return head->next;
        while(fast){
            prevslow= slow;
            slow = slow->next;
            fast = fast->next;

        }
        prevslow->next = slow->next;

        return head;
    
    }
};